test/compiler/codegen.jl

```diff
@@ -860,3 +860,16 @@ foo50964(1) # Shouldn't assert!
 # https://github.com/JuliaLang/julia/issues/51233
 obj51233 = (1,)
 @test_throws ErrorException obj51233.x
+
+# Very specific test for multiversioning
+if Sys.ARCH === :x86_64
+    foo52079() = Core.Intrinsics.have_fma(Float64)
+    if foo52079() == true
```

Comment:
would this test have caught the issue? I thought we were miscompiling `have_fma` also.

Reply:
Yes (this test fails in 1.10/master), `code_native` tends to lie. but `code_native(...dump_module=false)` doesn't lie too much since it just gets whateve the JIT has.

Reply:
We weren't miscompiling. We were just not multiversioning as we should.

Reply:
IIRC, it is considered a miscompilation in this case, since we defined that `have_fma` is not permitted to disagree with itself within any given run of the program (e.g. what happened in #52079).

Reply:
Doesn't it always do that? (if we are interpreting it returns false) i.e
```julia
julia> Core.Intrinsics.have_fma(Float64)
false

julia> foo() = Core.Intrinsics.have_fma(Float64)
foo (generic function with 2 methods)

julia> foo()
true
```

Reply:
Uh oh. That seems very bad

Reply:
I believe we intentionally made it return false from the global scope (but I forget why)

Reply:
Hm, seems like it was a known bug in the initial PR that was not fixed before merging https://github.com/JuliaLang/julia/pull/43085#discussion_r758714415

Reply:
I don't think there's anything semantically wrong with have_fma returning different values, but it is certainly undesirable.

Reply:
In any case I opened https://github.com/JuliaLang/julia/pull/52206